### ISSUE TITLE
False negative for `ClassStructure`

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#13295]
## Expected behavior

Offense

## Actual behavior

No offense

## Steps to reproduce the problem

```yaml
Layout/ClassStructure:
  Enabled: true
  ExpectedOrder:
    - public_attribute_macros
    - initializer
```

```ruby
class Foo
  def initialize
  end

  attr_reader :foo
end
```

Since the reader is after the initializer, RuboCop should complain.

## RuboCop version

```
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-md 1.2.2
  - rubocop-minitest 0.35.1
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.8.5
```

**Comment by Earlopain** [comment]
I think this is a configuration error. For this to work like explained in the [docs](https://docs.rubocop.org/rubocop/1.66/cops_layout.html#layoutclassstructure), you need to provide more data:

```yml
Layout/ClassStructure:
  Enabled: true
  ExpectedOrder:
    - module_inclusion
    - constants
    - association
    - public_attribute_macros
    - public_delegate
    - macros
    - public_class_methods
    - initializer
    - public_methods
    - protected_attribute_macros
    - protected_methods
    - private_attribute_macros
    - private_delegate
    - private_methods
  Categories:
    association:
      - has_many
      - has_one
    attribute_macros:
      - attr_accessor
      - attr_reader
      - attr_writer
    macros:
      - validates
      - validate
    module_inclusion:
      - include
      - prepend
      - extend
```

By default, it is not so sophisticated:
![image](https://github.com/user-attachments/assets/9e6395ef-095a-45ac-9602-6a15338e3d1a)

**Comment by andyw8** [comment]
Ah, thank you. If I copy and paste the config it does work. But I would have expected that that if put _only_ this in the config:

```
Layout/ClassStructure:
  Enabled: true
```

then it would use the [default configuration](Layout/ClassStructure:
  Enabled: true), but that seems to not be the case (it doesn't cause an offense)

**Comment by Earlopain** [comment]
It _does_ use that config, the cop just has unfortunate documentation. The only example is for the more elaborate config you need to write yourself (which is not the default).

Docs also assume that `delegate` respects visibility modifiers, which isn't the case. They are always public, unless you pass `private : true`.

Feel free to reopen this as a docs issue instead, looks like this could use some work.

**Comment by Okkun555** [comment]
@bbatsov 
I'd be happy to contribute a fix for this issue if no one is working on it yet.
